packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,233 @@
+    if (formattedCode) {
+      formattedCode = formattedCode.replaceAll(key, value);
+    }
+  }
+
+  return formattedCode;
+};
+
+$: {
+  if (element && code) {
+    element.innerHTML = formatCode(code);
+    highlight();
+  }
+}
+
+onMount(async () => {
+  try {
+    /**
+     * Load the themes, for vite to properly discover them we cannot use string templates,
+     * so we use a switch statement for our supported themes.
+     */
+    switch (theme) {
+      case 'vs': {
+        await import('prism-themes/themes/prism-vs.min.css');
+        break;
+      }
+      case 'vsc-dark-plus': {
+        await import('prism-themes/themes/prism-vsc-dark-plus.min.css');
+        break;
+      }
+      default: {
+        // Unsupported theme
+        break;
+      }
+    }
+
+    /**
+     * After the HTML is loaded in the DOM, we can use the autoloader to manage
+     * scanning for languages and including the components properly
+     */
+    await import(
+      // @ts-expect-error no type declaration for this JS file
+      'prismjs/plugins/autoloader/prism-autoloader'
+    );
+
+    // Make sure the autoloader knows where to find our languages
+    (Prism.plugins.autoloader as { languages_path: string }).languages_path =
+      // TODO: do we want a better way to get the version?
+      `https://cdnjs.cloudflare.com/ajax/libs/prism/${PRISM_VERSION}/components/`;
+
+    // Do the initial highlighting
+    if (element) {
+      Prism.highlightElement(element);
+    }
+  } catch (error) {
+    /* eslint-disable-next-line no-console */
+    console.error('Error initializing prismjs', error);
+  }
+});
+</script>
+
+<figure class={cx('flex flex-col gap-2', extraClasses)}>
+  {#if $$slots.default}
+    <figcaption><slot /></figcaption>
+  {/if}
+
+  <div
+    class={cx('flex gap-x-4 p-2', {
+      'bg-gray-9': theme === 'vsc-dark-plus',
+      'bg-light': theme === 'vs',
+    })}
+  >
+    <!-- The formatting here is intentional to preserve the formatting of `code` -->
+    <pre class="flex-1 overflow-x-auto"><code
+        bind:this={element}
+        class="language-{language}"
+        data-dependencies={dependencies.join(',')}>{code}</code
+      ></pre>
+
+    {#if showCopyButton}
+      <IconButton
+        class="text-black"
+        icon={copyButtonIcon}
+        label={copyButtonLabel}
+        on:click={copyToClipboard}
+        on:keyup={copyToClipboard}
+      />
+    {/if}
+  </div>
+</figure>
+
+{#if theme}
+  <style>
+  /* Theme overrides */
+  figure pre[class*='language-'],
+  figure pre[class*='language-'] > code[class*='language-'] {
+    margin: 0;
+    border: none;
+    background: inherit;
+  }
+
+  figure pre[class*='language-'] {
+    padding: 0;
+    padding-left: 0.5rem;
+    padding-top: 0.5rem;
+  }
+
+  figure pre[class*='language-'] > code[class*='language-'] {
+    font-family: 'Roboto Mono Variable', 'Roboto Mono', ui-monospace, monospace;
+  }
+  </style>
+{/if}
```

Comment:
Why can't these be added to the `class` normally? They all look like regular tailwind utilities

Reply:
They get overridden by the prism styles. We were using important modifiers on every tailwind class but forcing important felt worse than using more specific selectors. These are essentially a light “reset” of the prism css to not include backgrounds and borders, and use the correct padding and fonts.

If we feel it’s better to use tailing utilities with important modifiers then I can revert it.